Rdconverter/__init__.py

```diff
@@ -0,0 +1,159 @@
+import re
+import yaml as yam
+
+def curate_string(string):
+    string = string.replace(" ", "")
+    return string
+
+
+def yaml_type(string):
+    """ this function take as input the default value of an argument and return the type of it (option_writer)""""
+    liste = [
+    re.compile("False", re.IGNORECASE),
+    re.compile("true", re.IGNORECASE),
+    re.compile("[1-9]+"),
+    re.compile("[1-9]+\.[1-9]+")
+    ]
+    correspondance = ['boolean', 'boolean', 'string', 'string']
+    for element in liste:
+        if element.search(string) != None:
+            return correspondance[liste.index(element)]
+        else:
+            return 'string'
+
+class counting():
+"""this class take as self a string starting at the begigning of an argument explanation in a .Rd file :
+        \item{argument name }{ help string ..... 
+         ....
+         end of the .Rd file
+    The goal of this counting.count is too find where the help string end and return a raw string of it.
+    It is call counting because it counts the { and  } to determine where it ends 
+"""    
+    def __init__(self, liste, current=1, once=True, help_raw = ''):
+        self.current = current
+        self.liste = liste
+        self.once = once
+        self.help_raw = ''
+
+    def count(self):
+        for line in self.liste:
+
+            for char in line:
+                if not self.once:
+                    self.help_raw += char
+                    if char == "{":
+                        self.current += 1
+                if char == "}":
+                    self.current -= 1
+                if self.current == 0 and not self.once:
+                    return self.help_raw
+                if self.current == 0 and self.once:
+                    self.once = False
+
+def curate_default(string):
+    #string = string.replace(r"\\", r"\")
+    string = string.replace(" ", "")
+    string = string.replace(",", "")
+    return string
+
```

Comment:
You have several of these 'curate string' methods scattered around- could you have a single (parameterised for different purposes) function somewhere?

Reply:
I knew you would say that! yes I can with the one that do the same